tests/unit/sql_test.py

```diff
@@ -0,0 +1,302 @@
+        self.assertEqual(EXPECTED_UPDATE_COUNT, self.result.update_count().result())
+        self.assertFalse(self.result.is_row_set().result())
+
+        with self.assertRaises(ValueError):
+            self.result.get_row_metadata().result()
+
+        with self.assertRaises(ValueError):
+            for _ in self.result:
+                pass
+
+    def test_execute_error(self):
+        self.set_execute_error(RuntimeError("expected"))
+        with self.assertRaises(HazelcastSqlError) as cm:
+            iter(self.result)
+
+        self.assertEqual(_SqlErrorCode.GENERIC, cm.exception._code)
+
+    def test_execute_error_when_connection_is_not_live(self):
+        self.connection.live = False
+        self.set_execute_error(RuntimeError("expected"))
+        with self.assertRaises(HazelcastSqlError) as cm:
+            iter(self.result)
+
+        self.assertEqual(_SqlErrorCode.CONNECTION_PROBLEM, cm.exception._code)
+
+    def test_close_when_execute_is_not_done(self):
+        future = self.result.close()
+        self.set_close_response()
+        self.assertIsNone(future.result())
+        with self.assertRaises(HazelcastSqlError) as cm:
+            iter(self.result)
+
+        self.assertEqual(_SqlErrorCode.CANCELLED_BY_USER, cm.exception._code)
+
+    def test_close_when_close_request_fails(self):
+        future = self.result.close()
+        self.set_close_error(HazelcastSqlError(None, _SqlErrorCode.MAP_DESTROYED, "expected", None))
+
+        with self.assertRaises(HazelcastSqlError) as cm:
+            future.result()
+
+        self.assertEqual(_SqlErrorCode.MAP_DESTROYED, cm.exception._code)
+
+    def test_fetch_error(self):
+        self.set_execute_response_with_rows(is_last=False)
+        result = []
+        i = self.result.iterator().result()
+        # First page contains two rows
+        result.append(next(i).result().get_object_with_index(0))
+        result.append(next(i).result().get_object_with_index(0))
+
+        self.assertEqual(EXPECTED_ROWS, result)
+
+        # initiate the fetch request
+        future = next(i)
+
+        self.set_fetch_error(RuntimeError("expected"))
+
+        with self.assertRaises(HazelcastSqlError) as cm:
+            future.result()
+
+        self.assertEqual(_SqlErrorCode.GENERIC, cm.exception._code)
+
+    def test_fetch_server_error(self):
+        self.set_execute_response_with_rows(is_last=False)
+        result = []
+        i = self.result.iterator().result()
+        # First page contains two rows
+        result.append(next(i).result().get_object_with_index(0))
+        result.append(next(i).result().get_object_with_index(0))
+
+        self.assertEqual(EXPECTED_ROWS, result)
+
+        # initiate the fetch request
+        future = next(i)
+
+        self.set_fetch_response_with_error()
+
+        with self.assertRaises(HazelcastSqlError) as cm:
+            future.result()
+
+        self.assertEqual(_SqlErrorCode.PARSING, cm.exception._code)
```

Comment:
Same here I feel like this tests is not very useful due to the same reason

Reply:
Same, code coverage for error cases